api/lib/domain/usecases/compute-campaign-analysis.js

```diff
@@ -0,0 +1,52 @@
+const _ = require('lodash');
+const CampaignAnalysis = require('../models/CampaignAnalysis');
+const CampaignTubeRecommendation = require('../models/CampaignTubeRecommendation');
+const { UserNotAuthorizedToAccessEntity } = require('../errors');
+
+module.exports = async function computeCampaignAnalysis(
+  {
+    userId,
+    campaignId,
+    campaignRepository,
+    competenceRepository,
+    targetProfileRepository,
+    tubeRepository,
+  } = {}) {
+
+  const hasUserAccessToResult = await campaignRepository.checkIfUserOrganizationHasAccessToCampaign(campaignId, userId);
+
+  if (!hasUserAccessToResult) {
+    throw new UserNotAuthorizedToAccessEntity('User does not have access to this campaign');
+  }
+
+  const competences = await competenceRepository.list();
+  const tubes = await tubeRepository.list();
+
+  const targetProfile = await targetProfileRepository.getByCampaignId(campaignId);
+  const { targetedSkillsByTubeId } = _groupByColumn(targetProfile.skills, 'tubeId');
+
+  const campaignTubeRecommendations = _forgecampaignTubeRecommendations(campaignId, tubes, competences, targetedSkillsByTubeId);
+
+  return new CampaignAnalysis({ id: campaignId, campaignTubeRecommendations });
+};
+
+function _groupByColumn(targetedSkills, columnName) {
+  const targetedSkillsByTubeId = _.groupBy(targetedSkills, columnName);
+  return { targetedSkillsByTubeId };
```

Comment:
pourquoi le retourner dans un objet ?
```suggestion
  return targetedSkillsByTubeId;
```

Reply:
Je proposerais même de supprimer cette fonction, même dans son nom elle montre bien qu'elle n'est qu'un _wrapper_ de `_.groupBy`.

`const targetedSkillsByTubeId = _.groupBy(targetProfile.skills, 'tubeId');` me paraît déjà parfaitement explicite ?

Reply:
En fait `targetedSkillsByTubeId` est même inutile, on n'a d'intérêt que pour les `targetedTubeIds`, d'où le code par lequel j'ai remplacé tout ça.